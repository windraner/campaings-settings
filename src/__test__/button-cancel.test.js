import React from 'react';
import ButtonCancel from '../components/main-button-section/button-cancel/button-cancel';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<ButtonCancel />', () => {
  it('Check for className and text', () => {
    const wrapper = shallow(<ButtonCancel handlerCallBack={function() {}} />);
    expect(wrapper.find('.campaigns-settings__cancel-button').length).toBe(1);
    expect(wrapper.find('.campaigns-settings__cancel-button').text()).toBe('Cancel');
  });
  it('Check for callback is called', () => {
    const onClickFunc = jest.fn();
    const wrapper = shallow(<ButtonCancel handlerCallBack={onClickFunc} />);
    const button = wrapper.find('.campaigns-settings__cancel-button');
    button.simulate('click');
    expect(onClickFunc).toBeCalled();
  });
  it('matches the snapshot', () => {
    const tree = shallow(<ButtonCancel handlerCallBack={function() {}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
