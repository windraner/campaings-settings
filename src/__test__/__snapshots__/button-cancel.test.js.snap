// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCancel /> matches the snapshot 1`] = `
<div
  className="campaigns-settings__cancel-button"
  onClick={[Function]}
>
  Cancel
</div>
`;
